src/tests/functional/frame_navigation_tests.ts

```diff
@@ -24,6 +24,13 @@ test("test frame navigation with exterior link", async ({ page }) => {
   await nextEventOnTarget(page, "frame", "turbo:frame-load")
 })
 
+test("test frame navigation emits fetch-error event when offline", async ({ page }) => {
+  await page.goto("/src/tests/fixtures/tabs.html")
```

Comment:
While I appreciate that you're recreating the failure you experienced with the page that you experienced it on, is there a link in the `/src/tests/fixtures/frame_navigation.html` page that we could exercise for the same outcome? That way we could cut down on the number of `page.goto` calls.